app/src/main/java/dev/ukanth/ufirewall/Api.java

```diff
@@ -791,27 +792,32 @@ private static boolean applyIptablesRulesImpl(final Context ctx, RuleDataSet rul
                 cmds.add("-A " + AFWALL_CHAIN_NAME + " -o " + itf + " -j " + AFWALL_CHAIN_NAME + "-3g");
             }
 
-            final boolean any_wifi = ruleDataSet.wifiList.indexOf(SPECIAL_UID_ANY) >= 0;
-            final boolean any_3g = ruleDataSet.dataList.indexOf(SPECIAL_UID_ANY) >= 0;
-
-            // special rules to allow 3G<->wifi tethering
+            // special rules to allow tethering
             // note that this can only blacklist DNS/DHCP services, not all tethered traffic
-            if (((!whitelist && (any_wifi || any_3g)) ||
-                    (ruleDataSet.dataList.indexOf(SPECIAL_UID_TETHER) >= 0) || (ruleDataSet.wifiList.indexOf(SPECIAL_UID_TETHER) >= 0))) {
-
-                String[] users = {"root", "nobody"};
-                String action = " -j " + (whitelist ? "RETURN" : AFWALL_CHAIN_NAME + "-reject");
+            String[] users_dhcp = {"root", "nobody", "network_stack"};
```

Comment:
Why "network_stack" ?

Reply:
I saw it in logs: `Unknown(1073) 255.255.255.255:68`
`1073` is `network_stack`

Reply:
It might be recently introduced. You may want to add SDK check, because this will throw exceptions on lower sdk's

https://android.googlesource.com/platform/system/core/+/master/libcutils/include/private/android_filesystem_config.h#122

https://android.googlesource.com/platform/system/core/+/android-6.0.1_r5/include/private/android_filesystem_config.h

Reply:
> You may want to add SDK check, because this will throw exceptions on lower sdk's

Where? [`android.os.Process.getUidForName`](https://developer.android.com/reference/android/os/Process#getUidForName(java.lang.String)) will return -1 if user does not exists

Reply:
You are right, It was not during getUidForName , but while applying the rules using su.